apps/vaporgui/VStringLineEdit.cpp

```diff
@@ -0,0 +1,45 @@
+#include <string>
+#include <sstream>
+#include <iomanip>
+
+#include <QString>
+#include <QLineEdit>
+
+#include <VStringLineEdit.h>
+
+VStringLineEdit::VStringLineEdit( std::string value ) : 
+    VContainer(),
+    _lineEdit( new QLineEdit ),
+    _strValue( value )
+{
+    layout()->addWidget( _lineEdit );
+    connect( _lineEdit, SIGNAL( editingFinished() ),
+        this, SLOT( _valueChanged() ) );
+
+    _lineEdit->setContextMenuPolicy( Qt::DefaultContextMenu );
+    _lineEdit->setText( QString::fromStdString( value ) );
+    _lineEdit->setToolTip( QString::fromStdString( value ) );
+}
+
+void VStringLineEdit::SetValueString( std::string value ) {
+    _strValue = value;
+    _lineEdit->setText( QString::fromStdString( _strValue ) );
+    _lineEdit->setToolTip( QString::fromStdString( _strValue ) );
```

Comment:
The text input widgets set their tooltip to their value. Doesn’t this prevent us from giving them "help" tooltips?

Reply:
>Doesn’t this prevent us from giving them "help" tooltips?

No - the help tooltips will be assigned to the VLineItem, taking up the label and the space between the label and VLineEdit.  

Assigning a separate tooltip that shows the current text is helpful for showing long numbers and directory paths that aren't visible because of size constraints.